model/src/main/kotlin/DependencyGraphNavigator.kt

```diff
@@ -45,20 +45,21 @@ class DependencyGraphNavigator(
     override fun scopeNames(project: Project): Set<String> = project.scopeNames.orEmpty()
 
     override fun directDependencies(project: Project, scopeName: String): Sequence<DependencyNode> {
-        // TODO: Relax this assumption that package manager names start with the name of the type of project
-        //       they manage, for example that "GradleInspector" manages "Gradle" projects.
-        val managers = graphs.keys.filter { it.startsWith(project.id.type) }
+        // Collect all root indices for all managers whose graphs have projects of the respective type.
+        val rootIndicesForGraphs = graphs.mapNotNull { (manager, graph) ->
+            graph.scopes[DependencyGraph.qualifyScope(project, scopeName)]?.let { Triple(manager, graph, it) }
+        }
+
+        if (rootIndicesForGraphs.isEmpty()) return emptySequence()
 
-        val manager = requireNotNull(managers.singleOrNull()) {
+        // TODO: Consider changing the API so that this function takes a manager name and direct dependencies from
+        //       multiple managers for the same project type could be explicitly queried.
```

Comment:
Does this sentince miss something, e.g. like a comma? I miss how "could be explicitly queried" connect to the leading part.

Reply:
If a `managerName` was part of the signature of this function, then dependencies for that manager could be explicitly queried, without "guessing" which (of potentially multiple enabled managers) created e.g. the passed Gradle-type project.

Reply:
Thanks, to me the sentence as is is hard to understand, I believe it should be rephrased a bit. As I do not fully get the point, it's hard to make a proposal. Maybe something like the following?

Consider extending the signature to also take a manager name, so that the dependencies for that manager can be obtained without "guessing" which of the managers created the project.

Reply:
Fine with me.